### PR TITLE
[82, 69, 107, 62] `MentorshipCard`, `MenteeCard`, `RequestsCard`, and `ApplicationsCard` redesign

### LINKED ISSUE #62: RequestCard: short message, shrinks width
### Describe the bug

In Requests page, if the text of the message request is not long enough, the content of the request card does not fill the whole card.

### To Reproduce

1. Create a Mentorship Request with a short text (or edit one in the xata db)
1. Go to Requests page

### Screenshots

image

### LINKED ISSUE #69: Mentor: `MenteeCard` [redesign]
### Problem

In #68 the list of mentees and their cards was refactored to be the list of mentees in a relationship with the current mentor user.

## Dependencies

- #107 

### Feature Description

The `MenteeCard` should represent the mentee in a relationship with the mentor. 

- Redesign `MenteeCard` to show mentee info
- Render the mentorship relationship: show on the card `MentorshipInfo`
- This design should be consistent with the new `MentorshipCard`. See #82 .

followup ticket for the future:
- Clicking `Details` on the card, should lead to a page with the details of the `relationships` with the current mentor. See #70 .
- The `MentorshipDetails` in #70 embeds the `MenteeCard`, which in this case should not show the `Details` button.

### Screenshots

mentor-mentees-page

### LINKED ISSUE #82: Mentee: `MentorshipCard` [redesign]
### Problem

The current design of the card is not showing interesting info

## Dependencies

- #107 

### Feature Description

The `MentorshipCard` should represent the mentor/mentee relationship. 

- Redesign `MentorshipCard` to show mentor info
- Render the mentorship relationship: show on the card `MentorshipInfo`
- This design should be consistent with the new `MenteeCard`. See #69 .

followup ticket for the future:
- Redesign to render the relationship: show on the card, all the important details relative to the `relationship`.
- Clicking `Details` on the card, should lead to a page with the details of the `relationships` with the current mentor. See #70 .
- The `MentorshipDetails` in #70 embed the `MentorshipCard`, which in this case should not show the `Details` button.

### Screenshots

mentee-mentorships-page

### LINKED ISSUE #107: `MentorshipInfo` important mentorship details
### Feature Description

(See whimsical)

`MentorshipInfo` , all the important details relative to the `relationship`, something in a vertical layout.

This card is to be embedded into the 4 cards `MentorshipCard`, `MenteeCard`, `RequestsCard`, and `ApplicationsCard`.

1. Start date, or Date of the request.
2. Long/Short term.
3. Message of the request.
4. ~Button: "Book a call" OR "emailto:" if there is no link in the profile~ moved to `MentorshipCard`, and `MenteeCard`.
5. ~Button: Close Mentorship~ moved to `MentorshipCard`, and `MenteeCard`.

image

### PR DESCRIPTION
**Pull request by lorepirri**
### What Github issue does this PR relate to?

- Close #82 
- Close #69 
- Close #107
- Close #62

Changes proposed in this pull request:

- major redesign of `MentorshipCard`, `MenteeCard`, `RequestsCard`, and `ApplicationsCard`

### What should the reviewer know?

Try all.

### CODE REVIEW

File: components/MentorshipsList/MentorshipInfo.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        width='100%'
+      >
+      
+      <VStack
+          divider={<StackDivider borderColor='gray.200' />}
+          spacing={4}
+          align='stretch'
+      >
+        { (dateLabel && mentorshipDate) && 
+          <Text textAlign={"right"} mt={4}>{dateLabel}: {mentorshipDate}</Text>
```
Comment (maranta1) [inline components/MentorshipsList/MentorshipInfo.js:36]:
```suggestion
          <Text mt={4}>{dateLabel}: {mentorshipDate}</Text>
```

File: components/MentorshipsList/MentorshipInfo.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        }
+        <ReactMarkdown components={ChakraUIRenderer(markdownTheme)} skipHtml>
+          {message}
+        </ReactMarkdown>
+        <Text textAlign={"center"} mt={4} as='b'>
```
Comment (maranta1) [inline components/MentorshipsList/MentorshipInfo.js:41]:
```suggestion
        <Text mt={4} as='b'>
```

File: components/MenteesList/MenteeCard.js [hunk 3]
```diff
@@ -32,68 +21,37 @@ const MenteeCard = ({ mentorship }) => {
...
+            <MenuButton
+              as={Button}
+              rightIcon={<ChevronDownIcon />}
+            >
+              Close
+            </MenuButton>
+            <MenuList>
+              <MenuItem 
+                onClick={() => handleCloseMentorship(mentorship, RelationshipStatusEnum.ClosedCompleted)}
+              >Completed</MenuItem>
```
Comment (maranta1) [inline components/MenteesList/MenteeCard.js:43]: The `handleCloseMentorship` function is not defined yet, will this be created in another PR?

Comment (maranta1) [reply components/MenteesList/MenteeCard.js:43]: Ah we just need to pass it down as a prop from MenteesList/index.js

File: components/MentorshipsList/MentorshipInfo.js (hunk 1 above)
Comment (maranta1) [reply components/MentorshipsList/MentorshipInfo.js:36]: I think left aligning the content in the card could look better, WDYT?
Screenshot 2023-02-14 at 08 22 11
Screenshot 2023-02-14 at 08 22 02

File: components/MentorshipsList/MentorshipInfo.js (hunk 2 above)
Comment (maranta1) [reply components/MentorshipsList/MentorshipInfo.js:41]: Same as above

File: components/MentorshipsList/MentorshipInfo.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
+import { Stack, VStack, Text, StackDivider } from "@chakra-ui/layout";
+import { RequestStatusEnum, RelationshipStatusEnum } from "../../types/dbTablesEnums";
+import ChakraUIRenderer from "chakra-ui-markdown-renderer";
+import ReactMarkdown from "react-markdown";
+
+const markdownTheme = {
+  p: (props) => {
+    const { children } = props;
+    return <Text mb={2}>{children}</Text>;
```
Comment (maranta1) [inline components/MentorshipsList/MentorshipInfo.js:9]:
```suggestion
    return <Text>{children}</Text>;
```

File: components/MentorshipsList/MentorshipInfo.js (hunk 1 above)
Comment (maranta1) [reply components/MentorshipsList/MentorshipInfo.js:36]: We could also format the date using something like this: `new Intl.DateTimeFormat().format(new Date("2022-08-06T22:25:37.873Z"))`

File: components/MentorshipsList/MentorshipCard.js [hunk 5]
```diff
@@ -34,59 +22,54 @@ const MentorshipCard = ({ mentorship }) => {
...
+          ?
+          <Button 
+            rightIcon={<ExternalLinkIcon />} colorScheme='greenButton' variant='solid'
+            onClick={() => window.open(mentor?.mentor?.calendly)}
+          >
+            Book a Call
+          </Button>
+          :
+          (mentor?.email &&
+          <Button
```
Comment (maranta1) [inline components/MentorshipsList/MentorshipCard.js:45]:
```suggestion
                <Button
                  as="a"
                  leftIcon={<EmailIcon />} colorScheme='greenButton' variant='solid'
                  href={`mailto:${mentor.email}`}
                >
                  Email
                </Button>
```
We can use the `as` prop to get chakra-ui to render a link instead which is more accessible

File: components/MentorshipsList/MentorshipCard.js [hunk 6]
```diff
@@ -34,59 +22,54 @@ const MentorshipCard = ({ mentorship }) => {
...
+          leftIcon={<EmailIcon />} colorScheme='greenButton' variant='solid'
+          onClick={() => window.open('mailto:'+mentor?.email)}
+          >
+            Email
+          </Button>)}
```
Comment (maranta1) [inline components/MentorshipsList/MentorshipCard.js:50]:
```suggestion
                <Button
                  as="a"
                  leftIcon={<EmailIcon />} colorScheme='greenButton' variant='solid'
                  href={`mailto:${mentor.email}`}
                >)}
```
We can the use the `as` prop to get chakra-ui to render a link instead which is more accessible

File: components/MentorshipsList/MentorshipCard.js [hunk 7]
```diff
@@ -34,59 +22,54 @@ const MentorshipCard = ({ mentorship }) => {
...
           textAlign={"right"}
           justifyContent={"center"}
         >
-          <Button colorScheme={"greenButton"}>Details</Button>
+          {
+          !!mentor?.mentor?.calendly
+          ?
+          <Button 
+            rightIcon={<ExternalLinkIcon />} colorScheme='greenButton' variant='solid'
+            onClick={() => window.open(mentor?.mentor?.calendly)}
```
Comment (maranta1) [inline components/MentorshipsList/MentorshipCard.js:39]: We can use the `as` prop here to render a link as well (see comment below)

**Review by maranta1 — APPROVED** [review]
Great work! 🚀

File: components/MentorshipsList/MentorshipInfo.js (hunk 4 above)
Comment (maranta1) [reply components/MentorshipsList/MentorshipInfo.js:9]: See [comment](https://github.com/tupuio/tupuapp/pull/116/files#r1105441528)

File: components/MenteesList/MenteeCard.js (hunk 3 above)
Comment (lorepirri) [reply components/MenteesList/MenteeCard.js:43]: done

File: components/MentorshipsList/MentorshipCard.js (hunk 6 above)
Comment (lorepirri) [reply components/MentorshipsList/MentorshipCard.js]: done

File: components/MentorshipsList/MentorshipCard.js (hunk 7 above)
Comment (lorepirri) [reply components/MentorshipsList/MentorshipCard.js]: done

**Comment by lorepirri** [comment]
thanks @maranta1 for reviewing it! I've addressed all your comments, merging now

**Pull request merged**